File: internal/security/leak.go
package security

// Leak data struct
type Leak struct {
	File        string   `json:"file"`
	Line        string   `json:"line"`
	LineNumber  int      `json:"line_number"`
	Offender    string   `json:"offender"`
	Remediation string   `json:"remediation"`
	Rule        string   `json:"rule"`
	Severity    Severity `json:"severity"`
	Tags        []string `json:"tags,omitempty"`
}

File: web/src/models/leak.ts
import { Repository } from "./repository"
import { Rule } from "./rule"

export interface Leak {
  id: string,
  author_email: string,
  author_name: string,
  commit_hash: string,
  created_at: string,
  file: string,
  line: string,
  line_number: number,
  occurence: number,
  offender: string,
  repository: Repository,
  rule: Rule,
  severity: string
}
